quil-py/quil/program/__init__.pyi

```diff
@@ -238,6 +314,73 @@ class CalibrationSet:
         """
         ...
 
+
+@final
+class FixedScheduleItem:
```

Comment:
These classes could really benefit from some `__repr__()` or `__str__()` since the example code (including the update I suggested) above will just output the following when the schedule `.items()` is `print()`ed:

```
[<builtins.FixedScheduleItem object at 0x7fc726269440>, <builtins.FixedScheduleItem object at 0x7fc72626bd80>, <builtins.FixedScheduleItem object at 0x7fc7262694d0>, <builtins.FixedScheduleItem object at 0x7fc726269320>, <builtins.FixedScheduleItem object at 0x7fc72626b9f0>]
```

I think in this case `pyquil` can be a very thin wrapper and the majority of the functionality can reside here? How do you see division of responsibility between the two packages?

Reply:
When I do this now, I see:

```
[ComputedScheduleItem { time_span: TimeSpan { start_time: Seconds(3.44e-7), duration: Seconds(4e-8) }, instruction_index: 4 }, ComputedScheduleItem { time_span: TimeSpan { start_time: Seconds(9.600000000000001e-8), duration: Seconds(2.48e-7) }, instruction_index: 3 }, ComputedScheduleItem { time_span: TimeSpan { start_time: Seconds(5.6000000000000005e-8), duration: Seconds(4e-8) }, instruction_index: 1 }, ComputedScheduleItem { time_span: TimeSpan { start_time: Seconds(0.0), duration: Seconds(4e-8) }, instruction_index: 2 }, ComputedScheduleItem { time_span: TimeSpan { start_time: Seconds(0.0), duration: Seconds(5.6000000000000005e-8) }, instruction_index: 0 }]
```

I was expecting something close to an idiomatic output of a Python dataclass. Here I see the underlying Rust class name, which has some different member names and is missing some of the inferred members (like the end time). Can it be fixed?

Reply:
>  Can it be fixed?

Yes, but/and:

* ...but it's not new to this PR; that's how `impl_repr!` implements it for all of our Python-bound types across all relevant Rigetti packages
* ...and that could make it quick & easy to fix if that makes a difference

Reply:
Understood. I feel its confusing so definitely a nice-to-have but outside the scope of this feature (or library!).